README.md

```diff
@@ -420,152 +811,209 @@ A candidate will be selected if:
+
+* What is the role of `search_size` in candidate ranker?
+
+For a given query, DeezyMatch searches for candidates iteratively. At each iteration (i.e., one colored region in the figure below), the query vector is compared with five potential candidate vectors according to the selected ranking metric (`ranking_metric` argument). Here, `search_size` is five. In step-1, the five closest candidate vectors, as measured by L2-norm distance, are examined. Four (out of five) candidate vectors passed the threshold (specified by `selection_threshold` argument) in the figure (step-1). However, in this example, we assume `num_candidates` is 10. So, DeezyMatch examines the second batch of potential candidates, again five vectors (as specified by `search_size`). Three (out of five) candidates pass the threshold in step-2. Finally, in the third iteration, three more candidates are found. DeezyMatch collects the information of these ten candidates and go to the next query.
+
+This adaptive search algorithm significantly reduces the computation time to find and rank a set of candidates in a dataset. Instead of searching the whole dataset, DeezyMatch iteratively compares a query vector with the "most-promising" candidates.
+
+<p align="center">
+<img src="./figs/query_candidate_selection.png" alt="role of search_size in candidate ranker" width="70%">
+</p>
+
+* In most use cases, `search_size` can be set `>= num_candidates`. However, if `num_candidates` is very large, it is better to set the `search_size` to lower values. Let's clarify this in an example. First, assume `num_candidates=4` (number of desired candidates is 4 for each query). If we set the `search_size` to values less than 4, let's say, 2. DeezyMatch needs to do at least two iterations. In the first iteration, it looks at the closest 2 candidate vectors (as `search_size` is 2). In the second iteration, candidate vectors 3 and 4 will be examined. So two iterations. Another choice is `search_size=4`. Here, DeezyMatch looks at 4 candidates in the first iteration, if they pass the threshold, it is done. If not, it will seach candidates 5-8 in the next iteration. Now, let's assume `num_candidates=1001` (i.e., number of desired candidates is 1001 for each query). If we set the `search_size=1000`, DeezyMatch has to search at least 2000 candidates (2 x 1000 `search_size`). If we set `search_size=100`, this time, DeezyMatch has to search at least 1100 candidates (11 x 100 `search_size`). So 900 vectors less. In the end, it is a trade-off between iterations and `search_size`.
```

Comment:
Maybe instead of " it is done" you can say "the process concludes" ? not sure

Reply:
I think this final example works fine and is very clear, but I wouldn't keep it in a bullet point, as it's quite long. if you just remove the bullet point i think it will work better for readability

Reply:
Done: https://github.com/Living-with-machines/DeezyMatch/pull/68/commits/d96bbcaeb48e5204d2b2878df01c96205c894bb8

Reply:
Bullet point is removed here: https://github.com/Living-with-machines/DeezyMatch/pull/68/commits/fe588451857feba88c10e475d43e898231e148cf